DetectorDescription/DDCMS/interface/DDVolumeProcessor.h

```diff
@@ -0,0 +1,91 @@
+#ifndef DETECTOR_DESCRIPTION_DD_VOLUME_PROCESSOR_H
+#define DETECTOR_DESCRIPTION_DD_VOLUME_PROCESSOR_H
+
+#include "DD4hep/VolumeProcessor.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "DetectorDescription/DDCMS/interface/DDSpecParRegistry.h"
+
+#include <string>
+#include <string_view>
+#include <vector>
+
+namespace cms {
+  
+  class DDVolumeProcessor : public dd4hep::PlacedVolumeProcessor {
+  public:
+
+    using Volume = dd4hep::Volume;
+    using PlacedVolume = dd4hep::PlacedVolume;
+    using PlacedVolumeProcessor = dd4hep::PlacedVolumeProcessor;
+    
+    DDVolumeProcessor() = default;
+    ~DDVolumeProcessor() override = default;
+
+    std::string_view stripNamespace(std::string_view v) {
+      auto first = v.find_first_of(":");
+      v.remove_prefix(std::min(first+1, v.size()));
+      return v;
+    }
+    
+    std::string_view stripCopyNo(std::string_view v) {
+      auto found = v.find_last_of("_");
+      if(found != v.npos) {
+	v.remove_suffix(v.size() - found);
+      }
+      return v;
+    }
+    
+    std::vector<std::string_view> split(std::string_view str, const char* delims) {
+      std::vector<std::string_view> ret;
+  
+      std::string_view::size_type start = 0;
+      auto pos = str.find_first_of(delims, start);
+      while(pos != std::string_view::npos) {
+	if(pos != start) {
+	  ret.emplace_back(str.substr(start, pos - start));
+	}
+	start = pos + 1;
+	pos = str.find_first_of(delims, start);
+      }
+      if(start < str.length())
+	ret.emplace_back(str.substr(start, str.length() - start));
+      return ret;
+    }
+
+    bool compare(std::string_view s1, std::string_view s2) {
```

Comment:
Aren't these last four methods `const`? Actually, why are they in this class? I'm not understanding their relationship to the class.

Reply:
The idea was to use this class as a replacement for the DDFilteredView. Unfortunately, it turns out that performance-wise the base class recursive "process" has problems and is not as fast as TGeoIterator (e.g. DFS). 
So, I'm going to put it on ice until the developers will improve it and use ROOT API instead.